crates/ink/codegen/src/generator/metadata.rs

```diff
@@ -316,6 +358,55 @@ impl Metadata<'_> {
         }
     }
 
+    /// Generates ink! metadata for the given return type of a constructor.
+    /// If constructor result type is not `Result`,
+    /// the metadata will not display any metadata for return type.
+    /// Otherwise, the return type is `Result<(), E>`.
+    fn generate_constructor_return_type(ret_ty: Option<&syn::Type>) -> TokenStream2 {
+        /// Returns `true` if the given type is `Self`.
+        fn type_is_self_val(ty: &syn::Type) -> bool {
```

Comment:
Please move this directly into the match statement, I find it confusing. See https://github.com/paritytech/ink/pull/1473